packages/web/src/js/Toast.ts

```diff
@@ -258,21 +270,24 @@ class Toast extends BaseComponent {
       element?.setAttribute(ATTRIBUTE_ARIA_EXPANDED, 'false');
     });
 
-    this.element.classList.remove(CLASS_NAME_VISIBLE);
-    this.element.classList.add(CLASS_NAME_HIDDEN);
-    this.element.classList.add(CLASS_NAME_TRANSITIONING);
+    // Use the visibility style to check if the toast is already visually hidden
+    const { visibility } = getComputedStyle(this.element);
 
-    executeAfterTransition(
-      this.element,
-      () => {
-        EventHandler.trigger(this.element, Toast.eventName(EVENT_HIDDEN));
-        this.element.remove();
-      },
-      true,
-      SLOWEST_TRANSITION_PROPERTY_NAME as CSSProperties,
-    );
+    const finishHiding = () => {
+      EventHandler.trigger(this.element, Toast.eventName(EVENT_HIDDEN));
+      this.element.remove();
+      this.isShown = false;
```

Comment:
Is this line necessary? I think the instance is discarded once the element is removed from DOM.